### DISCUSSION TITLE
[ZK Stack] - Select the prover mode - NoProofs

### DISCUSSION THREAD

**Discussion by virgil2019** [zkSync-Community-Hub/zksync-developers#847] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.79.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I am trying to launch a ZK chain, how to set up a prover?

The configuration in the tutorial shows "Select the prover mode - NoProofs", I think it means not to run a prover, right?

And I some other questions
1. Can I use a validate contract deployed myself, instead of a shared contract?
2. How can I determine the layer of the ZK chain? Is my chain connected directly to L1, or to another ZK chain?

### Expected Behavior

I want to run a prover

### Current Behavior

Select the prover mode - NoProofs

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
Hello! Let's address your questions regarding setting up a prover for your ZK chain:

1. **Selecting the Prover Mode - NoProofs**: Yes, selecting "NoProofs" means that you are configuring the system to operate without running a prover. This mode is typically used for testing purposes where proof generation is not required, allowing the system to operate with much lower hardware requirements.

2. **Using a Custom Validate Contract**: Yes, you can deploy and use your own validate contract instead of a shared one. This allows for customization specific to your requirements.

3. **Determining the Layer of the ZK Chain**: To determine whether your ZK chain is connected directly to L1 or another ZK chain, you need to review the configuration settings related to the network connections and chain IDs in your setup. Typically, this information is specified during the initial configuration of your ZK chain environment.

If you want to run a prover, you will need to switch from the "NoProofs" mode to either a GPU or CPU-based prover, depending on your hardware capabilities. You can find detailed guidance on enabling and configuring the Boojum prover in the [ZKsync documentation on proving](https://docs.zksync.io/zk-stack/running/proving).

For more in-depth discussions and support, you can visit the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by kaymomin** [comment, marked as answer]
Hi @virgil2019 - yes! By default, your setup is without a prover and uses a `DummyExecutor` contract. You can enable the prover by doing the following:

To initialize the prover, first use the init command:

`zkstack prover init`

It will guide you through the necessary configuration.

Then to run the prover:

`zkstack prover run`

You can refer the docs [here.](https://docs.zksync.io/zk-stack/running/proving)

On knowing if your chain is connected directly to L1 - this is usually configured during the initial setup of your ZK chain.
